modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseXPath.java

```diff
@@ -184,6 +198,53 @@ public SynapseXPath(String xpathString) throws JaxenException {
     }
 
     /**
+     * Evaluate if the expression is compilable in XPath 2.0 format. This will only be used when its failing to
+     * compile in Jaxen
+     *
+     * @param xpathExpr XPath expression
+     * @param elem Expression Configurations
+     * @throws JaxenException will be thrown if the expression is malformed
+     */
+    public SynapseXPath(String xpathExpr, OMElement elem) throws JaxenException {
+        super(xpathExpr, SynapsePath.X_PATH, log);
+
+        /* Setting namespaces */
+        OMElementUtils.addNameSpaces(this, elem, log);
+        this.addNamespacesForFallbackProcessing(elem);
+        domXpath.setNamespaceContext(domNamespaceMap);
+
+        /* Setting Path type */
+        this.setPathType(SynapseXPath.X_PATH);
+
+        /* Compile and see if the expression is valid */
+        try {
+            XPathExpression expr = domXpath.compile(xpathExpr);
+        } catch (XPathExpressionException e) {
+            log.error("Error occurred while compiling XPath expression: " + xpathExpr);
+            throw new JaxenException(e);
+        }
+
+        /* Make evaluation of this expression forced to XPath 2.0 */
+        this.forceFailoverEvaluation = Boolean.TRUE;
+
+        PassThroughConfiguration conf = PassThroughConfiguration.getInstance();
+        bufferSizeSupport =conf.getIOBufferSize();
+
+        /* Handle special cases of content aware/unaware scenarios */
+        if (xpathExpr.contains("/")) {
+            contentAware = true;
+        } else if (xpathExpr.contains("get-property('From'") || xpathExpr.contains("get-property('FAULT')")) {
```

Comment:
we can merge if and else if statements to one since we only set contentAware = true;